api/src/client/dynamics/fire-safety/DynamicsFireSafetyInspectionClient.ts

```diff
@@ -0,0 +1,60 @@
+import { FireSafetyInspection } from "@client/dynamics/fire-safety/types";
+import { LogWriterType } from "@libs/logWriter";
+import axios, { AxiosError } from "axios";
+
+export interface FireSafetyInspectionClient {
+  getFireSafetyInspectionsByAddress: (
+    accessToken: string,
+    address: string
+  ) => Promise<FireSafetyInspection[]>;
+}
+export const DynamicsFireSafetyInspectionClient = (
+  logWriter: LogWriterType,
+  orgUrl: string
+): FireSafetyInspectionClient => {
+  const getFireSafetyInspectionsByAddress = async (
+    accessToken: string,
+    address: string
+  ): Promise<FireSafetyInspection[]> => {
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(`Dynamics Fire Safety Inspection Client - Id:${logId}`);
+
+    return axios
+      .get(
+        `${orgUrl}/api/data/v9.2/ultra_fireinspections?$select=createdon,ultra_inspectionended,ultra_numberofopenviolations,ultra_streetaddress&$filter=(ultra_streetaddress eq '${address}')&$top=10`,
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        }
+      )
+      .then((response) => {
+        logWriter.LogInfo(
+          `Dynamics Fire Safety Client - Id:${logId} - Response: ${JSON.stringify(response.data)}`
+        );
+        return response.data.value.map((response: DynamicsFireSafetyInspectionResponse) => {
+          return {
```

Comment:
[pebble] same as the other client file, can we pull the processing portion into it's own function so there is less going on in the function to improve readability